app/controllers/api/custom_buttons_controller.rb

```diff
@@ -6,8 +6,8 @@ def create_resource(_type, _id, data)
         custom_button = CustomButton.new(data.except("resource_action", "options"))
         custom_button.userid = User.current_user.userid
         custom_button.options = data["options"].deep_symbolize_keys if data["options"]
-        custom_button.create_resource_action!(data["resource_action"].deep_symbolize_keys) if data.key?("resource_action")
         custom_button.save!
+        custom_button.create_resource_action!(data["resource_action"].deep_symbolize_keys) if data.key?("resource_action")
```

Comment:
just curious, can the create_resource_action! method somehow fail, and we end up with a created Object, but never used/returned ?  do we need a transaction for the above two so the request is atomic ?

Reply:
I don't really think so. resource action create 1) has no validations and 2) are artifacts that link dialogs, resources, and automate, there's not much of a vector for failure

Reply:
yeah, probably similar to this one: https://github.com/ManageIQ/manageiq/pull/20013

@d-m-u can you review if this is required and do a followup PR?  This is great to fix the issue on rails 5.2

Reply:
We cross our streams at the same time and didn't use a mutex